python/fledge/services/core/server.py

```diff
@@ -1233,11 +1244,23 @@ async def get_service(cls, request):
 
     @classmethod
     async def get_auth_token(cls, request: web.Request) -> web.Response:
-        """ get oauth token
+        """ get auth token
 
             :Example:
                 curl -sX GET http://localhost:<core mgt port>/fledge/service/authtoken
         """
+        bearer_token = request.headers.get('bearer_token', "")
+        if bearer_token:
+            services_list = ServiceRegistry.all()
+            is_bearer_token_valid = False
+            for svc in services_list:
+                if svc._token == bearer_token:
```

Comment:
IMO, we should pass service name in HEADER as we don't have any service info in API route. So that we make sure that token is for that service only. Otherwise for the current scenario if passing bearer_token exists in service registry then TRUE irrespective of the service it is requesting with.

Reply:
request.headers.get('authorization', None) 
the bearer token better be a jwt token with refresh token and expiry; that bearer token will itself be used to identify service. if a service requires token it should return 401 if that does not match or empty.

In other words, 
service registration with name + single use token is similar to login (if arg parser has token=true in process script)
and then for next request bring that (similar to user' auth token) bearer token to identify the requester and allow to access resources.

Reply:
Sure, if we have an expiry we will need to make sure there is a mechanism to extend it or to issue new tokens as these services could run for a very long time.

Passing the service name in the header is not a bad idea, even if it is also encrypted int e token, that way we can do an extra level of checking. We can hide all of this inside the management client code.

Reply:
Yeah.
re: no explicit header required
JWT aud(ience) claim will take care of that. i.e. exactly, encrypted in the token